src/realm/sync/noinst/client_history_impl.cpp

```diff
@@ -800,27 +803,32 @@ void ClientHistory::update_sync_progress(const SyncProgress& progress, const std
     // Progress must never decrease
     if (progress.latest_server_version.version <
         version_type(root.get_as_ref_or_tagged(s_progress_latest_server_version_iip).get_as_int())) {
-        throw IntegrationException(ClientError::bad_progress, "latest server version cannot decrease");
+        throw IntegrationException(ErrorCodes::SyncProtocolInvariantFailed, "latest server version cannot decrease",
+                                   ProtocolError::bad_progress);
```

Comment:
Not technically related to these changes, but should we include the version numbers in the exception message, since we've seen some failures around this area recently?

Reply:
Have we seen these exceptions ever fire? We could add some more detail here.